k9mail/src/main/java/com/fsck/k9/helper/UnreadWidgetProperties.java

```diff
@@ -0,0 +1,140 @@
+package com.fsck.k9.helper;
+
+import android.content.Context;
+import android.content.Intent;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.AccountStats;
+import com.fsck.k9.BaseAccount;
+import com.fsck.k9.K9;
+import com.fsck.k9.Preferences;
+import com.fsck.k9.R;
+import com.fsck.k9.activity.FolderList;
+import com.fsck.k9.activity.MessageList;
+import com.fsck.k9.controller.MessagingController;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.search.LocalSearch;
+import com.fsck.k9.search.SearchAccount;
+
+public class UnreadWidgetProperties {
+
+    private int appWidgetId;
+    private String accountUuid;
+    private String folderName;
+    private Type type;
+
+    public UnreadWidgetProperties(int appWidgetId, String accountUuid, String folderName) {
+        this.appWidgetId = appWidgetId;
+        this.accountUuid = accountUuid;
+        this.folderName = folderName;
+        calculateType();
+    }
+
+    public String getTitle(Context context) {
+        String accountName = getAccount(context).getDescription();
+        switch (type) {
+            case SEARCH_ACCOUNT:
+            case ACCOUNT:
+                return accountName;
+            case FOLDER:
+                return context.getString(R.string.unread_widget_title, accountName, folderName);
+            default:
```

Comment:
No need to change this now.

The `default` case is unnecessary. The way to make the compiler happy, catch missing cases at runtime, and to make it possible to catch missing cases by static analysis, is to change the code to:

```java
switch (type) {
    …
}

throw new IllegalStateException("Missing case: " + type);
```